pandas/tests/tseries/frequencies/test_inference.py

```diff
@@ -267,7 +267,7 @@ def test_infer_freq_tz(tz_naive_fixture, expected, dates):
     ],
 )
 @pytest.mark.parametrize(
```

Comment:
this is not exactly the same as the OP (though it may have revelead the issue). this is a naive fixture (IOW the OP worked for naive & UTC) ,but NOT for other tzs.

so can also make this more comprehensive

Reply:
Not sure to get your comment yet ...

When I run the test `test_infer_freq_tz_transition`, it runs for a lot of tzs (None, UTC, US/Eastern, Asia/Tokyo, ...), for date_pairs that cover the DST changes (Fall, Spring and no change) and for freq = "H" (a.o. as it also tests for other infra-day frequencies). The test also refers to #8772 which is the issue I rephrased with a simple example in #39556.

My OP was only one case (tz=None, UTC, CET and freq=H) amongst these cases.
What was misleading in the original test is that the base frequency "H" that triggers the issue was not covered (probably because the author thought that testing with "3H" would cover "H" + other cases.

I can add a new test but I do not see what would differ from the current one (just my date_pairs would cover a full year which is not really needed for the test and my tz would be "CET" that is not covered yet other tz with DST are covered).

The small change in the test (adding freq="H" to the frequencies to test) breaks pandas before the bugfix.

I could adapt the current comment from "    # see gh-8772" to "    # see gh-8772 and gh-39556" to make it clearer ?

Reply:
sorry you are right, was misreading the fixture.